src/appkit.cs

```diff
@@ -27427,4 +27575,265 @@ interface NSStepperTouchBarItem
 		[Export ("customizationLabel")]
 		string CustomizationLabel { get; set; }
 	}
+
+	[Protocol]
+	interface NSTextInputTraits
+	{
+		[Export ("autocorrectionType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType AutocorrectionType { get; set; }
+
+		[Export ("spellCheckingType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType SpellCheckingType { get; set; }
+
+		[Export ("grammarCheckingType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType GrammarCheckingType { get; set; }
+
+		[Export ("smartQuotesType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType SmartQuotesType { get; set; }
+
+		[Export ("smartDashesType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType SmartDashesType { get; set; }
+
+		[Export ("smartInsertDeleteType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType SmartInsertDeleteType { get; set; }
+
+		[Export ("textReplacementType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType TextReplacementType { get; set; }
+
+		[Export ("dataDetectionType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType DataDetectionType { get; set; }
+
+		[Export ("linkDetectionType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType LinkDetectionType { get; set; }
+
+		[Export ("textCompletionType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType TextCompletionType { get; set; }
+	}
+
+	interface INSTextCheckingClient { }
+
+	[Protocol]
+	interface NSTextCheckingClient : NSTextInputTraits, NSTextInputClient
+	{
+		[Abstract]
+		[Export ("annotatedSubstringForProposedRange:actualRange:")]
+		[return: NullAllowed]
+		NSAttributedString AnnotatedSubstring (NSRange range, [NullAllowed] NSRange actualRange);
+
+		[Abstract]
+		[Export ("setAnnotations:range:")]
+		void SetAnnotations (NSDictionary<NSString, NSString> annotations, NSRange range);
+
+		[Abstract]
+		[Export ("addAnnotations:range:")]
+		void AddAnnotations (NSDictionary<NSString, NSString> annotations, NSRange range);
+
+		[Abstract]
+		[Export ("removeAnnotation:range:")]
+		void RemoveAnnotation (string annotationName, NSRange range);
+
+		[Abstract]
+		[Export ("replaceCharactersInRange:withAnnotatedString:")]
+		void ReplaceCharacters (NSRange range, NSAttributedString annotatedString);
+
+		[Abstract]
+		[Export ("selectAndShowRange:")]
+		void SelectAndShow (NSRange range);
+
+		[Abstract]
+		[Export ("viewForRange:firstRect:actualRange:")]
+		[return: NullAllowed]
+		NSView GetView (NSRange range, [NullAllowed] CGRect firstRect, [NullAllowed] NSRange actualRange);
```

Comment:
`[NullAllowed] CGRect firstRect` that sounded weird, since `CGRect` is a struct

https://developer.apple.com/documentation/appkit/nstextcheckingclient/3242726-viewforrange?language=objc
```
- (NSView *)viewForRange:(NSRange)range 
               firstRect:(NSRectPointer)firstRect 
             actualRange:(NSRangePointer)actualRange;
```

^ those are not the same types, you're hiding the pointer, which is not correct and that won't work at runtime.

If you mapped this in other API please update them ASAP.